### PR TITLE
test(blade): added jest-axe utility for a11y

### PR DESCRIPTION
**Pull request by anuraghazra**
- Added jest-axe 

TODO: 

React native a11y tests

### CODE REVIEW

File: packages/blade/src/_helpers/testing/assertAccessibe.native.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export default async function assertAccessible(
+  _container: unknown,
+  _axeOptions: unknown,
+): Promise<void> {
+  // TODO: implement this
```
[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/_helpers/testing/assertAccessibe.native.ts:5]: For native side. Jest-axe doesn't work on native, I tried out https://github.com/aryella-lacerda/react-native-accessibility-engine but no luck. It's failing all the tests for some reason. Couldn't get it working.

File: packages/blade/src/components/Typography/Heading/__tests__/Heading.web.test.tsx [hunk 2]
```diff
@@ -91,4 +92,9 @@ describe('<Heading />', () => {
       }
     }
   });
+
+  it('should be accessible', async () => {
+    const { container } = renderWithTheme(<Heading>Text content</Heading>);
+    await assertAccessible(container);
```
[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/components/Typography/Heading/__tests__/Heading.web.test.tsx:98]: Depends actually. It's best to have a separate test for accessibility only. Although you can also directly put assertAccessible in other tests.

File: packages/blade/src/_helpers/testing/assertAccessibe.native.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kamleshchandnani) [reply packages/blade/src/_helpers/testing/assertAccessibe.native.ts:5]: what do we do then finally?

[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/_helpers/testing/assertAccessibe.native.ts:5]: I'll suggest lets move ahead with web jest-axe only. We will relook about native later on. I might have to open a bug report on that react-native-a11y-engine repo.

File: packages/blade/src/components/Typography/Heading/__tests__/Heading.web.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Typography/Heading/__tests__/Heading.web.test.tsx:98]: Having a separate test seems okay to me, we could add more assertions if needed there later

File: packages/blade/src/_helpers/testing/assertAccessibe.web.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { axe } from 'jest-axe';
+import type { JestAxeConfigureOptions } from 'jest-axe';
+
+export default async function assertAccessible(
+  container: string | Element,
+  axeOptions?: JestAxeConfigureOptions,
+): Promise<void> {
+  const results = await axe(container, axeOptions);
+  expect(results).toHaveNoViolations();
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/_helpers/testing/assertAccessibe.web.ts:9]: This seems too vague to understand. What is the scope of this check? Like what all would it catch and what would it not?

**Review by kamleshchandnani — APPROVED** [review]

**Comment by kamleshchandnani** [comment]
Merging this for now as we need a base setup, we'll fine-tune it based on cases we come across later on

**Pull request merged**